src/main/java/com/educare/unitylend/controller/CommunityController.java

```diff
@@ -0,0 +1,56 @@
+package com.educare.unitylend.controller;
+
+import com.educare.unitylend.Exception.ControllerException;
+import com.educare.unitylend.model.Community;
+import com.educare.unitylend.service.CommunityService;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Slf4j
+@AllArgsConstructor
+@RestController
+@RequestMapping("/community")
+
+public class CommunityController extends BaseController {
+    CommunityService communityService;
+
+    @GetMapping("all-communities")
+    public List<Community> getAllCommunities() throws ControllerException {
+
+        try {
+            List<Community> communityList = communityService.getCommunities();
+            return communityList;
+        } catch (Exception e) {
+            log.error("Error encountered in getting the users");
+            throw new ControllerException("Error encountered in getting the users", e);
+        }
+    }
+    @GetMapping("get-community-by-tag")
+    public String getCommunityWithTag(@RequestParam String commonTag) throws ControllerException {
```

Comment:
Variable name would be fine with just `tag`, `commonTag ` is not making much sense